suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/VisitorStateModifications.java

```diff
@@ -114,12 +116,22 @@ public static Description interceptDescription(VisitorState visitorState, Descri
         // the error-prone checks it will then produce a replacement with all the checks suppressed.
         boolean alreadyReportedFix = FIXES.containsKey(firstSuppressibleParent);
 
-        SuppressingFix suppressingFix = FIXES.computeIfAbsent(
-                firstSuppressibleParent,
-                _ignored -> new SuppressingFix(
-                        Optional.ofNullable(visitorState.getSourceCode()), suppressWarnings, firstSuppressibleParent));
-
-        suppressingFix.addSuppression(description.checkName);
+        LazySuppressionFix suppressingFix = FIXES.computeIfAbsent(firstSuppressibleParent, _ignored -> {
+            // Initialize every fix with all remove-rollout: suppressions removed
+            Stream<String> existingSuppressions = suppressWarnings
+                    .map(AnnotationUtils::annotationStringValues)
+                    .orElseGet(Stream::of);
+            Set<String> existingNonRolloutSuppressions = existingSuppressions
+                    .filter(sup -> !sup.startsWith(CommonConstants.AUTOMATICALLY_ADDED_PREFIX))
```

Comment:
So the trick to avoid removing non-errorprone suppressions is to just handle `for-rollout` ones? I guess this is fine as we haven't yet added the feature that works on human authored suppressions - we can handle that functionality when we get to it in a later PR.